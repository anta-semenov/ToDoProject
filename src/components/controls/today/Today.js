import React from 'react'
import './Today.less'

export default class Today extends React.Component {
  render() {
    return (
      <div
        className={`today today--${this.props.appearance} ${this.props.checked ? 'is-checked' : ''} ${this.props.dimmed ? 'is-dimmed' : ''} ${this.props.disabled ? 'is-disabled' : ''}`}
        onClick={() => this.props.onClick()}
      />
    )
  }
}

Today.propTypes = {
  appearance: React.PropTypes.oneOf(['task-info', 'tasks-list', 'default']),
  checked: React.PropTypes.bool,
  dimmed: React.PropTypes.bool,
  disabled: React.PropTypes.bool,
  onClick: React.PropTypes.func.isRequired
}
Today.defaultProps = {
  appearance: 'default',
  checked: false,
  dimmed: false,
  disabled: false
}
