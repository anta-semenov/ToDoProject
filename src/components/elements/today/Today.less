@import '../../../styles/variables.less';

.today {
  position: relative;
  transition: @transition-basic;
  cursor: pointer;
  flex-shrink: 0;

  &::before,
  &::after {
    content: '';
    position: absolute;
    top: 4px;
    left: 4px;
    border-radius: 50%;
    pointer-events: none;
  }
  &::before {
    width: ~"calc(100% - 10px)";
    height: ~"calc(100% - 10px)";
    border: 1px solid @color-border-dark-gray;
    opacity: 1;
    transform: scale(1);
    transition: @transition-basic;
  }

  &::after {
    width: ~"calc(100% - 8px)";
    height: ~"calc(100% - 8px)";
    background-color: @color-orange;
    opacity: 0;
    transform: scale(0);
    transition: @transition-basic;
  }
  &:hover {
    &::before {
      border-color: @color-orange;
    }
  }
  &.is-checked {
    &::before {
      opacity: 0;
      transform: scale(0);
    }
    &::after {
      opacity: 1;
      transform: scale(1);
    }
  }
  &.is-disabled {
    pointer-events: none;
    opacity: 0.3;
  }
  &.is-dimmed {
    opacity: 0.3;
  }

  &.today--default {
    margin: 0;
    width: 16px;
    height: 16px;
  }
  &.today--task-info {
    margin: 0 0 8px 0;
    width: 16px;
    height: 16px;
  }
  &.today--tasks-list {
    margin: 0 6px 0 0;
    width: 16px;
    height: 16px;
  }
}

.today__svg {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: 2;
  border-radius: 50%;
  overflow: hidden;
  pointer-events: none;
}

.today__beams {
  stroke: @color-border-dark-gray;
  stroke-dasharray: 2 7;
  stroke-width: 8;
  transition: @transition-basic;
  transform-origin: center;
  transform: scale(1);
  animation: beams @transition-basic;

  .today:hover & {
    stroke: @color-orange;
  }

  &.is-checked {
    stroke-width: 0;
    transform: scale(0);
  }
}
.today__beams {
  stroke: @color-border-dark-gray;
  stroke-dasharray: 2 7;
  stroke-width: 8;
  transition: 0.5s ease-out 0.2s;
  transform-origin: center;

  .today:hover & {
    stroke: @color-orange;
  }
  .today.is-checked & {
    opacity: 0;
    transform: scale(2);
    transition: @transition-basic;
  }
}
.today__beams--active {
  stroke: @color-orange;
  stroke-dasharray: 2 7;
  stroke-width: 8;
  transform: scale(0);
  transition: @transition-basic;
  transform-origin: center;
  opacity: 0;

  .today.is-checked & {
    opacity: 1;
    transform: scale(1);
    transition: 0.5s ease-out 0.2s;
  }
}
@keyframes beams {
  0% {opacity: 1; transform: scale(1)}
  50% {opacity: 0; transform: scale(0)}
  51% {opacity: 0; transform: scale(2)}
  100% {opacity: 1; transform: scale(1)}
}
