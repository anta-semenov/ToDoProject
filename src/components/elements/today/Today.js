import React from 'react'
import './Today.less'

const Today = ({ appearance = 'default', checked = false, dimmed = false, disabled = false, onClick }) => (
  <div
    className={`today today--${appearance}${checked ? ' is-checked' : ''}${dimmed ? ' is-dimmed' : ''}${disabled ? ' is-disabled' : ''}`}
    onClick={() => onClick()}
  >
    <svg className='today__svg' viewBox='0 0 26 26' >
      <circle className='today__beams' cx='13' cy='13' r='13' fill='none' />
      <circle className='today__beams--active' cx='13' cy='13' r='13' fill='none' />
    </svg>
  </div>
)

Today.propTypes = {
  appearance: React.PropTypes.oneOf(['task-info', 'tasks-list', 'default']),
  checked: React.PropTypes.bool,
  dimmed: React.PropTypes.bool,
  disabled: React.PropTypes.bool,
  onClick: React.PropTypes.func.isRequired
}

export default Today
