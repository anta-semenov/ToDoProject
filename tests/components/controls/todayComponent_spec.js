import React from 'react'
import { shallow } from 'enzyme'
import Today from '../../../src/components/elements/today/Today'

describe('Today component', () => {
  test('Should render component with right default classes', () => {
    const closeBtn = shallow(<Today onClick={jest.fn()} />)
    expect(closeBtn).toMatchSnapshot()
  })
  test('Should render component with right classes for passed props', () => {
    const closeBtn1 = shallow(
      <Today checked={true} appearance={'task-info'} dimmed={true} onClick={jest.fn()} />
    )
    expect(closeBtn1).toMatchSnapshot()

    const closeBtn2 = shallow(
      <Today appearance={'tasks-list'} disabled={true} onClick={jest.fn()} />
    )
    expect(closeBtn2).toMatchSnapshot()
  })
})
