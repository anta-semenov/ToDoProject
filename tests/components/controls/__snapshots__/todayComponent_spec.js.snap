// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Today component Should render component with right classes for passed props 1`] = `
<div
  className="today today--task-info is-checked is-dimmed"
  onClick={[Function]}
>
  <svg
    className="today__svg"
    viewBox="0 0 26 26"
  >
    <circle
      className="today__beams"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
    <circle
      className="today__beams--active"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
  </svg>
</div>
`;

exports[`Today component Should render component with right classes for passed props 2`] = `
<div
  className="today today--tasks-list is-disabled"
  onClick={[Function]}
>
  <svg
    className="today__svg"
    viewBox="0 0 26 26"
  >
    <circle
      className="today__beams"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
    <circle
      className="today__beams--active"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
  </svg>
</div>
`;

exports[`Today component Should render component with right default classes 1`] = `
<div
  className="today today--default"
  onClick={[Function]}
>
  <svg
    className="today__svg"
    viewBox="0 0 26 26"
  >
    <circle
      className="today__beams"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
    <circle
      className="today__beams--active"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
  </svg>
</div>
`;
